src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -303,6 +303,24 @@ CreateDistributedInsertSelectPlan(Query *originalQuery,
 	return distributedPlan;
 }
 
+/*
+ * CreateInsertSelectIntoLocalTablePlan
+ */
+DistributedPlan *
+CreateInsertSelectIntoLocalTablePlan(uint64 planId, Query *originalQuery, PlannerRestrictionContext *plannerRestrictionContext)
+{
+	Query * selectQuery = ((RangeTblEntry *)lsecond(originalQuery->rtable))->subquery;
```

Comment:
You can use ExtractSelectRangeTableEntry to extract the select.

Reply:
Thanks 👍